server/gql/schema.js

```diff
@@ -4,24 +4,71 @@ import { dataSource } from "../models";
 
 const Schema = buildSchema(`
     enum DataSourceType {
-        InMemory
+        InMemory,
+        Postgres
     },
     type Query {
-        dataSources: [DataSource]
+        dataSources: [DataSource],
+        getOneDataSource(id: Int!): DataSource
     },
+    type Mutation {
+        createDataSource(name: String!, type: DataSourceType!,  config: String!): DataSource
+        deleteDataSource(id: Int!): DataSource
+        updateDataSource(id: Int!, name: String!, type: DataSourceType!,  config: String!): DataSource
+    },  
     type DataSource {
-        id: Int!       
+        id: Int!
         name: String!
         type: DataSourceType! 
         config: String!
     }
 `);
 
-function listDataSources() {
+const listDataSources = () => {
     info("listDataSources request");
     return dataSource.findAll();
-}
+};
 
-const root = { dataSources: listDataSources };
+const getOneDataSource = ({ id }) => {
+    info("getOneDataSource request");
+    return dataSource.findById(id);
+};
+
+const deleteDataSource = ({ id }) => {
+    info(`deleteDataSource request for id ${id}`);
+    return dataSource.findById(id)
+        .then(foundDataSource => {
+            if (!foundDataSource) {
+                return;
+            }
+            return foundDataSource.destroy({ force: true }); // eslint-disable-line
+        });
+};
+
+const updateDataSource = ({ id, name, type, config }) => {
+    info("updateDataSource request");
+    return dataSource.findById(id).then(foundDataSource => foundDataSource.update({
+        name,
+        type,
+        config
+    }));
+};
+
+const createDataSource = ({ name, type, config }) => {
```

Comment:
Is there any field validation?

Reply:
Graphql takes care of that. If that function is called it means the mutations has a valid format according to the schema.